### PR TITLE
Add code coverage using jacoco4sbt

### PR DESCRIPTION
**Pull request by marcospereira**
Additionally:

1. Change travis configuration to use cache and also execute jacoco task instead of just the test task
2. Upgrade Play to the last RC5

### CODE REVIEW

**Comment by marcospereira** [comment]
I'm adding code coverage to this project because it is the simplest one that we have, which basically means that it don't have subprojects. jacoco4sbt does not supports subprojects coverage aggregation yet. We also need to check if jacoco4sbt is a viable solution, since it is pretty stale right now (not even pull requests being merged).

Also, we need to decide where to show code coverage stuff and how it can be useful to us. My suggestion is to use something like [Codacy](https://www.codacy.com) or [Coveralls](https://coveralls.io/). A good thing about coveralls is that it have a [github bot](https://coveralls.zendesk.com/hc/en-us/articles/201774895-Notifications) that adds comments like this:

https://github.com/edulify/play-hikaricp.edulify.com/pull/60#issuecomment-86957471

WDYT?

**Comment by jroper** [comment]
@marcospereira best place to add code coverage would be in the interplay build plugins, by doing that, we can control the config for all Play projects in one place.

**Comment by marcospereira** [comment]
@jroper 

I thought about that, but as I said, jacoco4sbt is not working properly when there are subprojects:

1. It does not do code coverage aggregation between subprojects, so you have to look them individually to have an idea about how good is your code coverage
2. In some cases  I was not able to make it work (subprojects again). It could be just me, but I was not able to make it work with Play, per instance.
3. The project is pretty stale. There are PR from Sep 30, 2014 and they are piling up.

In other words, adding it to interplay could break a lot of our builds. So I decide to start small and see if this is useful to us. But, I agree with you that interplay is the best place. We just need to figure out how to bring some life to jacoco4sbt.

**Comment by Mogztter** [comment]
I think it's a good idea to add test coverage but I don't want to rely on a staled plugin.

What about https://github.com/scoverage ? Did you compare sbt-jacoco and sbt-scoverage ?

**Comment by marcospereira** [comment]
@Mogztter 

I just tried scoverage and it did not work for Play project. See the comment below to understand the problem:

https://github.com/scoverage/sbt-scoverage/issues/72#issuecomment-104503150

Besides that, the main point about using jacoco4sbt is that it can measure code coverage for both Scala and Java while scoverage cannot.

**Comment by jroper** [comment]
@marcospereira As far as I can tell (correct me if I'm wrong) the code coverage problems are only a problem if you run them (ie, problems only occur when your run `jacoco:cover`), otherwise, having the plugins installed/enabled, but never actually running the code coverage, doesn't cause a problem.  Is that right?  If so, then I think adding them to interplay would be good.

**Comment by marcospereira** [comment]
@jroper yes, you are right. If I understand your (implicit) point, we can add jacoco4sbt (or scoverage) to interplay and them execute the code coverage tasks in specifics projects only. This is totally doable.

Right now we have two main option:

1. We can use jacoco4sbt and help to move it forward if you guys know someone related to the project.
2. We can use scoverage, which does not report code coverage for Java code.

scoverage looks like a better solution for most of our projects, except for play itself, which has a lot of Java code.

**Comment by jroper** [comment]
scoverage does look better.... we could add both of them :)

**Comment by marcospereira** [comment]
Yep. I will close this in favor of opening a new PR at interplay.
